.travis.yml

```diff
@@ -1,12 +1,30 @@
 language: go
+os:
+  - windows
+  - linux
+  - osx
 env:
   - GO111MODULE=on
 go_import_path: github.com/rogpeppe/go-internal
+
+# Add this before_install until we have a definitive resolution on
+# https://travis-ci.community/t/files-in-checkout-have-eol-changed-from-lf-to-crlf/349/2
+before_install:
+  - cd ../..
+  - mv $TRAVIS_REPO_SLUG _old
+  - git config --global core.autocrlf false
+  - git clone --depth=50 _old $TRAVIS_REPO_SLUG
+  - cd $TRAVIS_REPO_SLUG
+
 install: "echo no install step required"
 go:
   - "1.11.x"
 script:
   - go test ./...
   - go mod tidy
-  - go list all > /dev/null # https://github.com/golang/go/issues/27868#issuecomment-431413621
+
+  # https://github.com/golang/go/issues/27868#issuecomment-431413621
+  - go list all > /dev/null
+
+  - test -z "$(gofmt -d .)" || (gofmt -d . && false)
   - test -z "$(git status --porcelain)" || (git status; git diff && false)
```

Comment:
tip - all beyond "go test ./..." doesn't need to be run on all platforms. In particular, Mac and Windows are way slower than a Linux container, so I tend to run all extra checks (gofmt, go mod, etc) on Linux only. For example: https://github.com/mvdan/sh/blob/master/.travis.yml#L11

Reply:
Thanks - this is good advice. Will make this change.

Reply:
Actually, I'm going to hold off this for now; would seem to make sense to see the line endings fiasco settle a bit first. I'm still scarred from those experiences!